scrapy/downloadermiddlewares/auth.py

```diff
@@ -28,19 +47,33 @@ def spider_opened(self, spider):
             self.auth = basic_auth_header(usr, pwd)
 
     def process_request(self, request, spider):
-        auth = getattr(self, 'auth', None)
-        if auth and 'Authorization' not in request.headers:
-            request.headers['Authorization'] = auth
```

Comment:
@umrashrf , it's less of a removal than moving statements around.
`getattr(self, 'auth', None)` is done a few lines after so that  `request.headers['Authorization']` is set only once and not overwritten.
Also `'Authorization' in request.headers` is relevant for HTTP only